code/test/tokens.js

```diff
@@ -0,0 +1,472 @@
+    // Alice Items that have changed - nonce, lastOperationTime, Commits, trackedTokens
+    assert.notStrictEqual(await alice.wallet.nonce, alice.state.nonce, 'alice wallet.nonce should have been changed')
+    // alice.state.nonce = parseInt(alice.state.nonce) + 1
+    alice.state.nonce = await alice.wallet.getNonce()
+    // lastOperationTime
+    assert.notStrictEqual(await alice.wallet.lastOperationTime(), alice.state.lastOperationTime, 'alice wallet.lastOperationTime should have been updated')
+    alice.state.lastOperationTime = await alice.wallet.lastOperationTime()
+    // commits
+    assert.notDeepEqual(await alice.wallet.getAllCommits(), alice.state.allCommits, 'alice wallet.allCommits should have been updated')
+    alice.state.allCommits = await alice.wallet.getAllCommits()
+    // tracked tokens
+    console.log(`alice.wallet.getTrackedTokens(): ${JSON.stringify(await alice.wallet.getTrackedTokens())}`)
+    console.log(`alice.state.trackedTokens: ${JSON.stringify(alice.state.trackedTokens)}`)
+    // assert.notDeepEqual(await alice.wallet.getTrackedTokens(), alice.state.trackedTokens, 'alice.wallet.trackedTokens should have been updated')
+    // alice.state.trackedTokens = await alice.wallet.getTrackedTokens()
+    await CheckUtil.checkONEWallet(alice.wallet, alice.state)
+    // Bob Items that have changed - tracked Tokens
+    console.log(`bob.wallet.getTrackedTokens(): ${JSON.stringify(await bob.wallet.getTrackedTokens())}`)
+    console.log(`bob.state.trackedTokens: ${JSON.stringify(bob.state.trackedTokens)}`)
+    assert.notDeepEqual(await bob.wallet.getTrackedTokens(), bob.state.trackedTokens, 'alice.wallet.trackedTokens should have been updated')
+    bob.state.trackedTokens = await bob.wallet.getTrackedTokens()
+    await CheckUtil.checkONEWallet(bob.wallet, bob.state)
+  })
+
+  // ERC1155 Testing (Transfer, Mint, Track) 
+  it('Wallet_CommitReveal: ERC1155(Transfer, Mint, Track) must commit and reveal successfully', async () => {
+    const alice = await makeWallet(1, accounts[1])
+    const aliceInitialWalletBalance = await web3.eth.getBalance(alice.wallet.address)
+    assert.equal(TEN_ETH, aliceInitialWalletBalance, 'Alice Wallet initially has correct balance')
+    const bob = await makeWallet(2, accounts[2])
+    const { testerc1155 } = await makeTokens(alice.lastResortAddress)
+    let aliceWalletBalanceERC1155T8
+    let bobWalletBalanceERC1155T8
+    assert.equal(20, await testerc1155.balanceOf(alice.lastResortAddress, 8), 'Alice.lastResortAddress owns 20 of token 8')
+    // transfer ERC721 tokens from alice.lastResortAddress (which owns the tokens) to alices wallet
+    // TODO review the bytes value we are passing in safeTransferFrom (currently using ONEUtil.hexStringToBytes('5') )
+    await testerc1155.safeTransferFrom(alice.lastResortAddress, alice.wallet.address, 8, 8, ONEUtil.hexStringToBytes('5'), { from: alice.lastResortAddress })
```

Comment:
5 is not really a hexString

Reply:
Have updated to use a constant for a dummy string
`const FIVE_HEX = ONEUtil.hexString('5')` which is `0x05`